--- dia-canvas-view-item.c	2003-03-18 19:21:22.000000000 +0100
+++ ../../diacanvas2-0.9.2.old/src/dia-canvas-view-item.c	2003-07-19 11:18:50.000000000 +0200
@@ -368,7 +368,7 @@
 		gnome_canvas_item_raise (GNOME_CANVAS_ITEM (vitem), positions);
 
 	/* Both items should be on the same position now. */
-	g_assert (dia_canvas_groupable_pos (item->parent, item) == g_list_index (GNOME_CANVAS_GROUP (GNOME_CANVAS_ITEM (vitem)->parent)->item_list, vitem));
+	g_assert (dia_canvas_groupable_pos (DIA_CANVAS_GROUPABLE (item->parent), item) == g_list_index (GNOME_CANVAS_GROUP (GNOME_CANVAS_ITEM (vitem)->parent)->item_list, vitem));
 }
 
 static void
@@ -1002,7 +1002,7 @@
 	/* Move the newly created DiaCanvasViewItem to the right place in
 	 * the item list (but not for the root item ;-). */
 	if (item->parent) {
-		item_index = dia_canvas_groupable_pos (item->parent, item);
+		item_index = dia_canvas_groupable_pos (DIA_CANVAS_GROUPABLE (item->parent), item);
 		vitem_index = g_list_index (vitem->item_list, new_vitem);
 		
 		/* New items are always added on the top, so if the
@@ -1018,13 +1018,13 @@
 
 	/* If item itself has children, add them also */
 	if (DIA_IS_CANVAS_GROUPABLE (item)
-	    && dia_canvas_groupable_get_iter (item, &iter)) do {
-		DiaCanvasItem *val = dia_canvas_groupable_value (item, &iter);
+	    && dia_canvas_groupable_get_iter (DIA_CANVAS_GROUPABLE (item), &iter)) do {
+		DiaCanvasItem *val = dia_canvas_groupable_value (DIA_CANVAS_GROUPABLE (item), &iter);
 
 		if (val)
 			dia_canvas_view_item_add_items (GNOME_CANVAS_GROUP (new_vitem),
 							val);
-	} while (dia_canvas_groupable_next (item, &iter));
+	} while (dia_canvas_groupable_next (DIA_CANVAS_GROUPABLE (item), &iter));
 }
 
 static gint
