--- dia-canvas-group.c	2003-01-14 19:41:42.000000000 +0100
+++ ../../diacanvas2-0.9.2.old/src/dia-canvas-group.c	2003-07-19 11:15:19.000000000 +0200
@@ -128,7 +128,7 @@
 
 	/* unref all children */
 	while (group->children)
-		dia_canvas_groupable_remove (DIA_CANVAS_ITEM (group),
+		dia_canvas_groupable_remove (DIA_CANVAS_GROUPABLE (group),
 					     group->children->data);
 
 	G_OBJECT_CLASS (parent_class)->dispose (object);
@@ -344,7 +344,7 @@
 		DIA_CANVAS_ITEM (group)->canvas->allow_undo = old_allow_undo;
 
 	if (!item->parent)
-		dia_canvas_groupable_add (DIA_CANVAS_ITEM (group), item);
+		dia_canvas_groupable_add (DIA_CANVAS_GROUPABLE (group), item);
 
 	/* We unref the item since the 'add' method did ref it. If it was
 	 * not ref'ed we destroy it this way... */
