--- test-item.c	2002-12-29 09:34:14.000000000 +0100
+++ ../../diacanvas2-0.9.2.old/unit_tests/test-item.c	2003-07-19 11:27:56.000000000 +0200
@@ -217,7 +217,7 @@
 {
 	TestItem *test_item = TEST_ITEM (object);
 	while (test_item->children)
-		dia_canvas_groupable_remove (DIA_CANVAS_ITEM (object),
+		dia_canvas_groupable_remove (DIA_CANVAS_GROUPABLE (object),
 					     test_item->children->data);
 
 	G_OBJECT_CLASS (parent_class)->dispose (object);
