--- dia-canvas.c	2003-03-18 19:31:50.000000000 +0100
+++ ../../diacanvas2-0.9.2.old/src/dia-canvas.c	2003-07-19 11:02:30.000000000 +0200
@@ -651,9 +651,9 @@
 
 	/* ... And now for the children... */
 	if (DIA_IS_CANVAS_GROUPABLE (item)
-	    && dia_canvas_groupable_get_iter (item, &iter)) do {
+	    && dia_canvas_groupable_get_iter (DIA_CANVAS_GROUPABLE (item), &iter)) do {
 		ArtDRect rdest;
-		DiaCanvasItem *child = dia_canvas_groupable_value (item, &iter);
+		DiaCanvasItem *child = dia_canvas_groupable_value (DIA_CANVAS_GROUPABLE (item), &iter);
 		GList *new_objs;
 		gdouble inv[6];
 
@@ -663,7 +663,7 @@
 							   &rdest);
 		if (new_objs)
 			objs = g_list_concat (objs, new_objs);
-	} while (dia_canvas_groupable_next (item, &iter));
+	} while (dia_canvas_groupable_next (DIA_CANVAS_GROUPABLE (item), &iter));
 
 	return objs;
 }
