--- dia-textbox.c	2003-03-02 15:32:43.000000000 +0100
+++ ../../diacanvas2-0.9.2.old/demos/dia-textbox.c	2003-07-19 11:25:45.000000000 +0200
@@ -137,7 +137,7 @@
 dia_textbox_init (DiaTextbox *textbox)
 {
 	textbox->text = dia_canvas_item_create (DIA_TYPE_CANVAS_TEXT, NULL);
-	dia_canvas_groupable_add_construction (DIA_CANVAS_ITEM (textbox),
+	dia_canvas_groupable_add_construction (DIA_CANVAS_GROUPABLE (textbox),
 						textbox->text);
 	//dia_canvas_item_request_update (textbox->text);
 	g_message (__FUNCTION__": %p %p %p %p %p", textbox, textbox->text, textbox->text->parent, DIA_CANVAS_ITEM (textbox)->canvas, textbox->text->canvas);
