--- dia-storage.h.~1.4.~	2001-08-01 17:24:51.000000000 +0200
+++ dia-storage.h	2003-08-06 09:56:26.000000000 +0200
@@ -133,9 +133,12 @@
 
 gboolean	dia_storage_save		(DiaStorage *storage);
 
-/* Dump the XML tree to a memory buffer, mainly for debugging. */
+/* Dump the XML tree to a memory buffer, mainly for debugging. 
+ *
+ * size: out parameter
+ */
 gchar*          dia_storage_to_buffer           (DiaStorage *storage,
-						 /*out*/gint *size);
+						 gint *size);
 
 /*
  * Object handling
@@ -173,11 +176,13 @@
 
 /* Name is also save in returned_arg->name. 
  * Returns `value' on success, NULL on error.
+ *
+ * value: in and out parameter
  */
 gboolean	dia_storage_attribute_get_value	(DiaStorage *storage,
 						 DiaAttributeNode *attr,
 						 DiaPersistent *object,
-						 /*< INOUT >*/GtkArg *value);
+						 GtkArg *value);
 
 /* Retrieve (child-)attributes from object or attribute. */
 DiaAttributeNode* dia_storage_get_attribute	(DiaStorage *storage,
