Index: dia-canvas-item.h
===================================================================
RCS file: /cvsroot/diacanvas/diacanvas2/src/dia-canvas-item.h,v
retrieving revision 1.1
diff -u -r1.1 dia-canvas-item.h
--- dia-canvas-item.h	15 Jul 2003 18:46:37 -0000	1.1
+++ dia-canvas-item.h	23 Aug 2003 10:47:28 -0000
@@ -278,6 +278,13 @@
 
 GType	dia_canvas_item_get_type	(void);
 
+/* Those methods can be used for compound objects. */
+void	dia_canvas_item_add_construction	(DiaCanvasItem *parent,
+						 DiaCanvasItem *item);
+void	dia_canvas_item_remove_destruction	(DiaCanvasItem *parent,
+						 DiaCanvasItem *item);
+
+
 DiaCanvasItem* dia_canvas_item_create	(GType type,
 					 const gchar *first_arg_name, ...);
 
