? removal_of_persistens.diff
Index: dia-canvas-bezier.c
===================================================================
RCS file: /cvsroot/diacanvas/diacanvas2/src/dia-canvas-bezier.c,v
retrieving revision 1.5
diff -u -r1.5 dia-canvas-bezier.c
--- dia-canvas-bezier.c	25 Mar 2002 15:44:53 -0000	1.5
+++ dia-canvas-bezier.c	13 Sep 2003 14:19:40 -0000
@@ -74,16 +74,9 @@
 static void
 dia_canvas_bezier_class_init (DiaCanvasBezierClass *klass)
 {
-#ifdef USE_PERSISTENT
-	static guint saveable_args[] = { ARG_BEZIER_WIDTH, ARG_COLOR, ARG_CAP, ARG_JOIN, 0 };
-#endif
 	GObjectClass *object_class;
 	DiaCanvasItemClass *item_class;
 	
-#ifdef USE_PERSISTENT
-	((DiaPersistentClass*) klass)->saveable_args = saveable_args;
-#endif
-  
 	object_class = (GObjectClass*) klass;
 	//item_class = (DiaCanvasItemClass*) klass;
 	item_class = DIA_CANVAS_ITEM_CLASS (klass);
Index: dia-canvas-element.c
===================================================================
RCS file: /cvsroot/diacanvas/diacanvas2/src/dia-canvas-element.c,v
retrieving revision 1.35
diff -u -r1.35 dia-canvas-element.c
--- dia-canvas-element.c	20 Jul 2003 14:14:21 -0000	1.35
+++ dia-canvas-element.c	13 Sep 2003 14:19:41 -0000
@@ -80,16 +80,9 @@
 static void
 dia_canvas_element_class_init (DiaCanvasElementClass *klass)
 {
-#ifdef USE_PERSISTENT
-	static guint saveable_args[] = { ARG_WIDTH, ARG_HEIGHT, 0 };
-#endif
 	GObjectClass *object_class;
 	DiaCanvasItemClass *item_class;
 	
-#ifdef USE_PERSISTENT
-	((DiaPersistentClass*) klass)->saveable_args = saveable_args;
-#endif
-  
 	object_class = (GObjectClass*) klass;
 	item_class = DIA_CANVAS_ITEM_CLASS (klass);
 	
Index: dia-canvas-ellipse.c
===================================================================
RCS file: /cvsroot/diacanvas/diacanvas2/src/dia-canvas-ellipse.c,v
retrieving revision 1.6
diff -u -r1.6 dia-canvas-ellipse.c
--- dia-canvas-ellipse.c	11 Feb 2002 15:40:54 -0000	1.6
+++ dia-canvas-ellipse.c	13 Sep 2003 14:19:41 -0000
@@ -228,8 +228,6 @@
 	GtkObjectClass *object_class;
 	DiaCanvasItemClass *item_class;
 	
-	((DiaPersistentClass*) klass)->saveable_args = saveable_args;
-  
 	object_class = (GtkObjectClass*) klass;
 	//item_class = (DiaCanvasItemClass*) klass;
 	item_class = DIA_CANVAS_ITEM_CLASS (klass);
Index: dia-canvas-line.c
===================================================================
RCS file: /cvsroot/diacanvas/diacanvas2/src/dia-canvas-line.c,v
retrieving revision 1.43
diff -u -r1.43 dia-canvas-line.c
--- dia-canvas-line.c	20 Jul 2003 14:14:21 -0000	1.43
+++ dia-canvas-line.c	13 Sep 2003 14:19:41 -0000
@@ -135,16 +135,9 @@
 static void
 dia_canvas_line_class_init (DiaCanvasLineClass *klass)
 {
-#ifdef USE_PERSISTENT
-	static guint saveable_args[] = { ARG_LINE_WIDTH, ARG_COLOR, ARG_CAP, ARG_JOIN, 0 };
-#endif
 	GObjectClass *object_class;
 	DiaCanvasItemClass *item_class;
 	
-#ifdef USE_PERSISTENT
-	((DiaPersistentClass*) klass)->saveable_args = saveable_args;
-#endif
-  
 	object_class = (GObjectClass*) klass;
 	item_class = DIA_CANVAS_ITEM_CLASS (klass);
 	
Index: dia-handle.c
===================================================================
RCS file: /cvsroot/diacanvas/diacanvas2/src/dia-handle.c,v
retrieving revision 1.39
diff -u -r1.39 dia-handle.c
--- dia-handle.c	19 May 2003 17:32:24 -0000	1.39
+++ dia-handle.c	13 Sep 2003 14:19:41 -0000
@@ -21,9 +21,6 @@
 #include <stdlib.h>
 #include <math.h>
 #include <libart_lgpl/art_affine.h>
-#ifdef USE_PERSISTENT
-#include "dia-storage.h"
-#endif
 #include "diamarshal.h"
 #include "diatypebuiltins.h"
 #include "dia-canvas-i18n.h"
@@ -59,13 +56,6 @@
 static void dia_handle_finalize		(GObject *object);
 static void pos_w_changed		(DiaVariable *variable,
 					 DiaHandle *handle);
-#ifdef USE_PERSISTENT
-static gboolean dia_handle_load (DiaStorage *storage, DiaAttributeNode *attr,
-				 DiaPersistent *object, GtkArg *value);
-static gboolean dia_handle_save (DiaStorage *storage, DiaAttributeNode *attr,
-				 DiaPersistent *object, GtkArg *value);
-static void dia_handle_free (GtkArg *value);
-#endif
 
 static GObjectClass *parent_class = NULL;
 //static guint handle_signals[LAST_SIGNAL] = { 0 };
@@ -113,11 +103,6 @@
 
 	parent_class = g_type_class_peek_parent (klass);
 
-#ifdef USE_PERSISTENT
-	dia_storage_register_type (dia_handle_get_type (),
-				   dia_handle_load, dia_handle_save,
-				   dia_handle_free);
-#endif
 	object_class->dispose = dia_handle_dispose;
 	object_class->finalize = dia_handle_finalize;
 	object_class->set_property = dia_handle_set_property;
