Index: ChangeLog
===================================================================
RCS file: /cvsroot/diacanvas/diacanvas2/ChangeLog,v
retrieving revision 1.135
diff -u -r1.135 ChangeLog
--- ChangeLog	9 Oct 2003 07:50:14 -0000	1.135
+++ ChangeLog	9 Oct 2003 10:55:25 -0000
@@ -1,3 +1,13 @@
+2003-10-09  Martin Willemoes Hansen  <mwh@sysrq.dk>
+
+	* src/dia-canvas.[ch]: Changed dia_canvas_new_pango_layout to dia_canvas_get_pango_layout
+	and registred it as a property.
+	* src/dia-shape-art.[ch]: Changed dia_shape_art_new_pango_layout to dia_shape_art_get_pango_layout.
+	* src/dia-shape-x.[ch]: Changed dia_shape_x_new_pango_layout to dia_shape_x_get_pango_layout and
+	fixed a typo.
+	* src/dia-shape.c: Updated to use dia_canvas_get_pango_layout
+	* src/diacanvas.defs: Ditto
+
 2003-10-04  Arjan Molenaar <arjanmolenaar@hetnet.nl>
 
 	* python/setup.py: New file. Build the python bindings in a Pythonic
Index: src/dia-canvas.c
===================================================================
RCS file: /cvsroot/diacanvas/diacanvas2/src/dia-canvas.c,v
retrieving revision 1.62
diff -u -r1.62 dia-canvas.c
--- src/dia-canvas.c	1 Sep 2003 18:26:43 -0000	1.62
+++ src/dia-canvas.c	9 Oct 2003 10:55:26 -0000
@@ -52,7 +52,8 @@
 	PROP_GRID_OFS_X,
 	PROP_GRID_OFS_Y,
 	PROP_GRID_COLOR,
-	PROP_GRID_BG
+	PROP_GRID_BG,
+	PROP_PANGO_LAYOUT
 };
 
 typedef struct {
@@ -264,6 +265,13 @@
 						 G_MAXULONG,
 						 DIA_COLOR (200,200,200),
 						 G_PARAM_READWRITE));
+	g_object_class_install_property (object_class,
+					 PROP_PANGO_LAYOUT,
+					 g_param_spec_object ("pango_layout",
+						 _("A newly created pango layout"),
+						 _("A newly created pango layout"),
+						 PANGO_TYPE_LAYOUT,
+						 G_PARAM_READABLE));
 
 	klass->extents_changed = NULL;
 	//klass->move = NULL;
@@ -407,6 +415,9 @@
 	case PROP_GRID_BG:
 		g_value_set_ulong (value, canvas->grid_bg);
 		break;
+	case PROP_PANGO_LAYOUT:
+		g_value_set_object (value, dia_canvas_get_pango_layout());
+		break;
 
 	default:
 		G_OBJECT_WARN_INVALID_PROPERTY_ID (object, property_id, pspec);
@@ -887,14 +898,14 @@
 } 
 
 /**
- * dia_canvas_new_pango_layout:
+ * dia_canvas_get_pango_layout:
  *
  * Returns a usable #PangoLayout object. 
  *
  * Return value: A newly created PangoLayout object.
  **/
 PangoLayout*
-dia_canvas_new_pango_layout (void)
+dia_canvas_get_pango_layout (void)
 {
 	/* Make the context static, this will save time for future references.
 	 */
Index: src/dia-canvas.h
===================================================================
RCS file: /cvsroot/diacanvas/diacanvas2/src/dia-canvas.h,v
retrieving revision 1.65
diff -u -r1.65 dia-canvas.h
--- src/dia-canvas.h	15 Jul 2003 18:46:37 -0000	1.65
+++ src/dia-canvas.h	9 Oct 2003 10:55:26 -0000
@@ -171,7 +171,7 @@
 void	dia_canvas_add_constraint	(DiaCanvas *canvas, DiaConstraint *c);
 void	dia_canvas_remove_constraint	(DiaCanvas *canvas, DiaConstraint *c);
 
-PangoLayout* dia_canvas_new_pango_layout (void);
+PangoLayout* dia_canvas_get_pango_layout (void);
 
 void	dia_canvas_redraw_views		(DiaCanvas *canvas);
 
Index: src/dia-shape-art.c
===================================================================
RCS file: /cvsroot/diacanvas/diacanvas2/src/dia-shape-art.c,v
retrieving revision 1.49
diff -u -r1.49 dia-shape-art.c
--- src/dia-shape-art.c	20 Jul 2003 14:14:21 -0000	1.49
+++ src/dia-shape-art.c	9 Oct 2003 10:55:26 -0000
@@ -70,17 +70,17 @@
 }
 
 /**
- * dia_shape_art_new_pango_layout:
+ * dia_shape_art_get_pango_layout:
  *
  * Return a new PangoLayout belonging to the AA renderer.
  *
  * Return value: A newly created PangoLayout object.
  **/
 PangoLayout*
-dia_shape_art_new_pango_layout (void)
+dia_shape_art_get_pango_layout (void)
 {
 	/* Use the default (canvas) PangoLayout setup. */
-	return dia_canvas_new_pango_layout ();
+	return dia_canvas_get_pango_layout ();
 }
 
 static ArtSVP*
@@ -679,7 +679,7 @@
 //	} else {
 //	}
 
-	layout = dia_shape_art_new_pango_layout ();
+	layout = dia_shape_art_get_pango_layout ();
 	max_width = (gint) ceil (shape->max_width);
 	max_height = (gint) ceil (shape->max_height);
 	dia_shape_text_fill_pango_layout ((DiaShape*) shape,
Index: src/dia-shape-art.h
===================================================================
RCS file: /cvsroot/diacanvas/diacanvas2/src/dia-shape-art.h,v
retrieving revision 1.7
diff -u -r1.7 dia-shape-art.h
--- src/dia-shape-art.h	9 Oct 2002 13:17:25 -0000	1.7
+++ src/dia-shape-art.h	9 Oct 2003 10:55:26 -0000
@@ -36,7 +36,7 @@
 void	dia_shape_art_render	(DiaShape *shape, DiaCanvasViewItem *item,
 				 GnomeCanvasBuf *buf);
 
-PangoLayout* dia_shape_art_new_pango_layout (void);
+PangoLayout* dia_shape_art_get_pango_layout (void);
 
 G_END_DECLS
 
Index: src/dia-shape-x.c
===================================================================
RCS file: /cvsroot/diacanvas/diacanvas2/src/dia-shape-x.c,v
retrieving revision 1.15
diff -u -r1.15 dia-shape-x.c
--- src/dia-shape-x.c	3 Jun 2003 18:08:38 -0000	1.15
+++ src/dia-shape-x.c	9 Oct 2003 10:55:26 -0000
@@ -361,9 +361,9 @@
 }
 
 PangoLayout*
-dia_shape_x_new_pango_layout (void)
+dia_shape_x_get_pango_layout (void)
 {
-	g_warning (G_STRLOC": No default fonr layout has been set...");
+	g_warning (G_STRLOC": No default font layout has been set...");
 	return NULL;
 }
 
Index: src/dia-shape-x.h
===================================================================
RCS file: /cvsroot/diacanvas/diacanvas2/src/dia-shape-x.h,v
retrieving revision 1.5
diff -u -r1.5 dia-shape-x.h
--- src/dia-shape-x.h	9 Oct 2002 13:17:25 -0000	1.5
+++ src/dia-shape-x.h	9 Oct 2003 10:55:26 -0000
@@ -38,7 +38,7 @@
 				 int x, int y,
 				 int width, int height);
 
-PangoLayout* dia_shape_x_new_pango_layout (void);
+PangoLayout* dia_shape_x_get_pango_layout (void);
 
 
 G_END_DECLS
Index: src/dia-shape.c
===================================================================
RCS file: /cvsroot/diacanvas/diacanvas2/src/dia-shape.c,v
retrieving revision 1.41
diff -u -r1.41 dia-shape.c
--- src/dia-shape.c	9 Oct 2003 07:50:15 -0000	1.41
+++ src/dia-shape.c	9 Oct 2003 10:55:26 -0000
@@ -1250,7 +1250,7 @@
 	g_return_val_if_fail (shape != NULL, NULL);
 	g_return_val_if_fail (shape->type == DIA_SHAPE_TEXT, NULL);
 
-	layout = dia_canvas_new_pango_layout ();
+	layout = dia_canvas_get_pango_layout ();
 	g_return_val_if_fail (layout != NULL, NULL);
 
 	if (fill)
Index: src/diacanvas.defs
===================================================================
RCS file: /cvsroot/diacanvas/diacanvas2/src/diacanvas.defs,v
retrieving revision 1.2
diff -u -r1.2 diacanvas.defs
--- src/diacanvas.defs	6 May 2002 09:43:54 -0000	1.2
+++ src/diacanvas.defs	9 Oct 2003 10:55:26 -0000
@@ -385,9 +385,9 @@
   )
 )
 
-(define-function dia_canvas_new_pango_layout
+(define-function dia_canvas_get_pango_layout
   (in-module "Dia")
-  (c-name "dia_canvas_new_pango_layout")
+  (c-name "dia_canvas_get_pango_layout")
   (return-type "PangoLayout*")
 )
 
@@ -2071,9 +2071,9 @@
   )
 )
 
-(define-function dia_shape_art_new_pango_layout
+(define-function dia_shape_art_get_pango_layout
   (in-module "Dia")
-  (c-name "dia_shape_art_new_pango_layout")
+  (c-name "dia_shape_art_get_pango_layout")
   (return-type "PangoLayout*")
 )
 
@@ -2095,9 +2095,9 @@
   )
 )
 
-(define-function dia_shape_x_new_pango_layout
+(define-function dia_shape_x_get_pango_layout
   (in-module "Dia")
-  (c-name "dia_shape_x_new_pango_layout")
+  (c-name "dia_shape_x_get_pango_layout")
   (return-type "PangoLayout*")
 )
 
