Index: .cvsignore
===================================================================
RCS file: /cvsroot/diacanvas/diacanvas2/unit_tests/.cvsignore,v
retrieving revision 1.1
diff -u -r1.1 .cvsignore
--- .cvsignore	16 May 2002 14:10:59 -0000	1.1
+++ .cvsignore	24 Aug 2003 15:41:52 -0000
@@ -2,4 +2,13 @@
 .deps
 Makefile
 Makefile.in
-
+test-canvas
+test-canvas-item
+test-canvas-view
+test-constraint
+test-expression
+test-geometry
+test-shape-art
+test-solver
+test-strength
+test-variable
