? GnomePrintFixes.diff
Index: ChangeLog
===================================================================
RCS file: /cvs/public/gtk-sharp/ChangeLog,v
retrieving revision 1.484
diff -u -r1.484 ChangeLog
--- ChangeLog	5 Oct 2003 00:37:24 -0000	1.484
+++ ChangeLog	8 Oct 2003 07:12:04 -0000
@@ -1,3 +1,12 @@
+2003-10-06  Martin Willemoes Hansen  <mwh@sysrq.dk>
+
+	* gnome/PrintJob.custom: Added default ctor.
+	* gnome/PrintDialog.custom: Added overloaded ctor with fewer parameters.
+	* sources/Gnome.metadata: Disabled default ctor for PrintJob,
+	  changed type from int to PrintDialogFlags for PrintDialog ctor and
+	  changed return type from byte to string for PrintConfig.Get.
+	* api/Makefile: Added gtkhtml-api.xml 
+
 2003-10-04  Mike Kestner  <mkestner@ximian.com>
 
 	* generator/CallbackGen.cs : remove CloseWriter call.
Index: api/gnome-api.xml
===================================================================
RCS file: /cvs/public/gtk-sharp/api/gnome-api.xml,v
retrieving revision 1.17
diff -u -r1.17 gnome-api.xml
--- api/gnome-api.xml	2 Oct 2003 15:48:36 -0000	1.17
+++ api/gnome-api.xml	8 Oct 2003 07:12:07 -0000
@@ -4937,7 +4937,7 @@
         </parameters>
       </method>
       <method name="Get" cname="gnome_print_config_get">
-        <return-type type="guchar*"/>
+        <return-type type="gchar*"/>
         <parameters>
           <parameter type="const-gchar*" name="key"/>
         </parameters>
@@ -5056,7 +5056,7 @@
         </parameters>
       </constructor>
     </object>
-    <object name="PrintJob" cname="GnomePrintJob" parent="GObject">
+    <object name="PrintJob" cname="GnomePrintJob" parent="GObject" disabledefaultconstructor="1">
       <method name="Close" cname="gnome_print_job_close">
         <return-type type="gint"/>
       </method>
@@ -6512,7 +6512,7 @@
         <parameters>
           <parameter type="GnomePrintJob*" name="gpj"/>
           <parameter type="const-gchar*" name="title"/>
-          <parameter type="gint" name="flags"/>
+          <parameter type="GnomePrintDialogFlags" name="flags"/>
         </parameters>
       </constructor>
       <method name="SetCopies" cname="gnome_print_dialog_set_copies">
Index: gnome/PrintDialog.custom
===================================================================
RCS file: gnome/PrintDialog.custom
diff -N gnome/PrintDialog.custom
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ gnome/PrintDialog.custom	8 Oct 2003 07:12:10 -0000
@@ -0,0 +1,12 @@
+//
+// PrintDialog.custom - customizations
+//
+// Author: Martin Willemoes Hansen <mwh@sysrq.dk>
+//
+// Copyright (C) 2003 Martin Willemoes Hansen
+//
+// This code is inserted after the automatically generated code.
+//
+	
+public PrintDialog (Gnome.PrintJob gpj, string title) 
+	: this (gpj, title, 0) {}
Index: gnome/PrintJob.custom
===================================================================
RCS file: gnome/PrintJob.custom
diff -N gnome/PrintJob.custom
--- /dev/null	1 Jan 1970 00:00:00 -0000
+++ gnome/PrintJob.custom	8 Oct 2003 07:12:10 -0000
@@ -0,0 +1,14 @@
+//
+// PrintJob.custom - customizations
+//
+// Author: Martin Willemoes Hansen <mwh@sysrq.dk>
+//
+// Copyright (C) 2003 Martin Willemoes Hansen
+//
+// This code is inserted after the automatically generated code.
+//
+	
+public PrintJob()
+{
+	Raw = gnome_print_job_new (PrintConfig.Default().Handle);
+}
Index: sources/Gnome.metadata
===================================================================
RCS file: /cvs/public/gtk-sharp/sources/Gnome.metadata,v
retrieving revision 1.12
diff -u -r1.12 Gnome.metadata
--- sources/Gnome.metadata	29 Sep 2003 11:05:30 -0000	1.12
+++ sources/Gnome.metadata	8 Oct 2003 07:12:10 -0000
@@ -319,6 +319,17 @@
   </data>
 </rule>
 
+<!-- disable default constructor -->
+<rule>
+  <class name="GnomePrintJob"/>
+  <data>
+    <attribute target="class">
+      <name>disabledefaultconstructor</name>
+      <value>1</value>
+    </attribute>
+  </data>
+</rule>
+
 <!-- misc -->
 <rule>
   <class name="GnomeAbout">
@@ -332,6 +343,18 @@
     </attribute>
  </data>
 </rule>
+<rule>
+  <class name="GnomePrintDialog">
+    <method>gnome_print_dialog_new</method>
+  </class>
+  <data>
+    <attribute target="params">
+      <filter level="name">flags</filter>
+      <name>type</name>
+      <value>GnomePrintDialogFlags</value>
+    </attribute>
+  </data>
+</rule>
 
 <!-- Use const-gchar* instead of const-guchar*, when the value is realy a string -->
 <rule>
@@ -499,6 +522,18 @@
     <attribute target="return">
       <name>type</name>
       <value>const-gchar*</value>
+    </attribute>
+  </data>
+</rule>
+
+<rule>
+  <class name="GnomePrintConfig">
+    <method>Get</method>
+  </class>
+  <data>
+    <attribute target="return">
+      <name>type</name>
+      <value>gchar*</value>
     </attribute>
   </data>
 </rule>
